presto-main/src/main/java/com/facebook/presto/execution/AlterFunctionTask.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.security.AccessControl;
+import com.facebook.presto.spi.function.AlterRoutineCharacteristics;
+import com.facebook.presto.spi.function.QualifiedFunctionName;
+import com.facebook.presto.spi.function.RoutineCharacteristics;
+import com.facebook.presto.spi.type.TypeSignature;
+import com.facebook.presto.sql.analyzer.Analyzer;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.tree.AlterFunction;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.transaction.TransactionManager;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.util.concurrent.Futures.immediateFuture;
+import static java.util.Objects.requireNonNull;
+
+public class AlterFunctionTask
+        implements DataDefinitionTask<AlterFunction>
+{
+    private final SqlParser sqlParser;
+
+    @Inject
+    public AlterFunctionTask(SqlParser sqlParser)
+    {
+        this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "ALTER FUNCTION";
+    }
+
+    @Override
+    public String explain(AlterFunction statement, List<Expression> parameters)
+    {
+        return "ALTER FUNCTION " + statement.getFunctionName();
+    }
+
+    @Override
+    public ListenableFuture<?> execute(AlterFunction statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
+    {
+        Analyzer analyzer = new Analyzer(stateMachine.getSession(), metadata, sqlParser, accessControl, Optional.empty(), parameters, stateMachine.getWarningCollector());
+        analyzer.analyze(statement);
+
+        QualifiedFunctionName functionName = QualifiedFunctionName.of(statement.getFunctionName().toString());
+        AlterRoutineCharacteristics alterRoutineCharacteristics = new AlterRoutineCharacteristics(
+                statement.getCharacteristics().getNullCallClause().map(nullCallClause -> RoutineCharacteristics.NullCallClause.valueOf(nullCallClause.name())));
```

Comment:
I know we are only supporting the null-call stuff but may be keep it open for other things so if we add something else it becomes easier to add.

Reply:
@kaikalur Yes, I agree. When we support altering function for other routine characteristics, we can add stuff into `AlterRoutineCharacteristics` accordingly. But I'm not sure why you're commenting on this line. Are you suggesting any specific changes?

Reply:
You are unconditionally processing only NullClause from AlterRoutine. There is no check that that's the only thing being changed. I was expecting that to be passed into this function as a param.

Reply:
> @kaikalur can you take a look at the syntax? Thanks!

Syntax part looks fine.

Reply:
The above line does not process any alter function or change any routine characteristics. What it is doing is to convert
`sql.tree.AlterRoutineCharacteristics` to `spi.function.AlterRoutineCharacteristics`. Neither class can take a reference of another class because neither presto-spi nor presto-parser are depending on another.

It is the `FunctionNamespaceManager`'s job to actually alter a routine given a `spi.function.AlterRoutineCharacteristics`.